### ISSUE TITLE
tf2 estimator and training strategies

### ISSUE THREAD

**Issue by davidsebfischer** [theislab/batchglm#92]
- [x] train_sequence cannot be defined here but has to be used as defined in models/base/estimator.py 
https://github.com/theislab/batchglm/blob/8f025e2fd6c985a17193e4601e7ceae091222f0a/batchglm/train/tf2/base_glm/estimator.py#L64
- [x] tf2 training strategies have to be adapted as discussed

**Comment by picciama** [comment]
`train_sequence` deleted
`trainingStrategies.py` deleted and integrated into model/estimator.
All changes within `train/tf2`.
@davidsebfischer If you checked whether your `train_sequence` method is working with our `train` method already, we are fine, otherwise I need to check that before we can close here. 
`train_sequence` calls `train` multiple times, if more than one optimizer is given. However, each time `train` is executed, a new model is created again in our case. So this isn't yet working, instead, the training will start from scratch for each given optimizer. 
Please tell me, if I need to fix this before closing here.

**Comment by davidsebfischer** [comment]
Please check that the new global train_sequence works for you, then close here.

You can close this even if train sequence restarts every time but open a new issue for the train sequence please.

**Comment by picciama** [comment]
train_sequence is working fine now. We might need to include additional information into th Enums of `training_strategies.py` in the future.
